packages/web/src/components/gcds-stepper/gcds-stepper.tsx

```diff
@@ -17,12 +17,33 @@ export class GcdsStepper {
   /**
    * Defines the current step.
    */
-  @Prop() currentStep!: number;
+  @Prop({ mutable: true }) currentStep!: number;
+  @Watch('currentStep')
+  validateCurrentStep() {
+    if (this.currentStep == undefined) {
+      this.currentStep = 1;
+    } else if (this.currentStep > this.totalSteps) {
+      this.currentStep = this.totalSteps;
```

Comment:
Just tested this and seems like this gets overwritten by the `totalSteps` validation. For example if `totalSteps` is set to `4` and I try to set `currentStep` to 28, it keeps `currentStep` as 28 and also updates `totalSteps` to 28.

Reply:
This is a tricky thing to figure out since they both have to validate independently. I guess we need to decide which attribute we want to change in a scenario where the current step is higher than the total steps.

Do we want `currentSteps` to be lowered down to the `totalSteps` value or have `totalSteps` raised up to the `currentStep` value?

Reply:
I'd vote lowering `currentStep` for `totalSteps` but I have no reasoning for it so I'm very much good with either :D

Reply:
Good catch Mel! If `currentStep` is out of bounds I would prefer to error out instead of trying to fix the user's (developer's) issue. I would also keep the range between 1 and x, we'll never have negative steps. So psuedocode something like
```javascript
if(this.currentStep === undefined){
  this.currentStep = 1;
} else if (this.currentStep < 1 || this.currentStep > this.totalSteps) { 
  console.error("Invalid number for curentStep"); or throw an error
}
```

Don't want to cause additional bugs for the user by setting the value to something else (apart from auto-increments?)

Commenting this as I haven't tested the entire code yet -- will do so soon!